src/library_openal.js

```diff
@@ -1,1622 +1,4542 @@
+      case 0x1006 /* AL_VELOCITY */:
+        return AL.currentCtx.audioCtx.listener._velocity;
+      case 0x100F /* AL_ORIENTATION */:
+        return AL.currentCtx.audioCtx.listener._direction.concat(AL.currentCtx.audioCtx.listener._up);
+      case 0x100A /* AL_GAIN */:
+        return AL.currentCtx.gain.gain.value;
+      default:
+#if OPENAL_DEBUG
+        console.error(funcname + "() param 0x" + param.toString(16) + " is unknown or not implemented");
+#endif
+        AL.currentCtx.err = 0xA002 /* AL_INVALID_ENUM */;
+        return null;
+      }
+    },
+
+    listenerHelper: function(funcname, param, value) {
+      if (!AL.currentCtx) {
+#if OPENAL_DEBUG
+        console.error(funcname + "() called without a valid context");
+#endif
+        return;
+      }
+      if (value === null) {
+#if OPENAL_DEBUG
+        console.error(funcname + "(): param 0x" + param.toString(16) + " has wrong signature");
+#endif
+        AL.currentCtx.err = 0xA002 /* AL_INVALID_ENUM */;
+        return;
+      }
+
+      var listener = AL.currentCtx.audioCtx.listener;
+      switch (param) {
+      case 0x1004 /* AL_POSITION */:
+        if (!Number.isFinite(value[0]) || !Number.isFinite(value[1]) || !Number.isFinite(value[2])) {
+#if OPENAL_DEBUG
+          console.error(funcname + "() param AL_POSITION value " + value + " is out of range");
+#endif
+          AL.currentCtx.err = 0xA003 /* AL_INVALID_VALUE */;
+          return;
+        }
+
+        listener._position = value;
+        AL.updateListenerSpace(AL.currentCtx);
+        break;
+      case 0x1006 /* AL_VELOCITY */:
+        if (!Number.isFinite(value[0]) || !Number.isFinite(value[1]) || !Number.isFinite(value[2])) {
+#if OPENAL_DEBUG
+          console.error(funcname + "() param AL_VELOCITY value " + value + " is out of range");
+#endif
+          AL.currentCtx.err = 0xA003 /* AL_INVALID_VALUE */;
+          return;
+        }
+
+        listener._velocity = value;
+        AL.updateListenerSpace(AL.currentCtx);
+        break;
+      case 0x100A /* AL_GAIN */:
+        if (!Number.isFinite(value) || value < 0.0) {
+#if OPENAL_DEBUG
+          console.error(funcname + "() param AL_GAIN value " + value + " is out of range");
+#endif
+          AL.currentCtx.err = 0xA003 /* AL_INVALID_VALUE */;
+          return;
+        }
+
+        AL.currentCtx.gain.gain.value = value;
+        break;
+      case 0x100F /* AL_ORIENTATION */:
+        if (!Number.isFinite(value[0]) || !Number.isFinite(value[1]) || !Number.isFinite(value[2])
+          || !Number.isFinite(value[3]) || !Number.isFinite(value[4]) || !Number.isFinite(value[5])
+        ) {
+#if OPENAL_DEBUG
+          console.error(funcname + "() param AL_ORIENTATION value " + value + " is out of range");
+#endif
+          AL.currentCtx.err = 0xA003 /* AL_INVALID_VALUE */;
+          return;
+        }
+
+        listener._direction = value.slice(0, 3);
+        listener._up = value.slice(3, 6);
```

Comment:
Removed the slicing here since the destination arrays are already guaranteed to exist.